components/raftstore-v2/src/operation/query/mod.rs

```diff
@@ -282,10 +282,20 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
                 && read.cmds()[0].0.get_requests().len() == 1
                 && read.cmds()[0].0.get_requests()[0].get_cmd_type() == CmdType::ReadIndex;
 
+            let read_index = read.read_index.unwrap();
             if is_read_index_request {
                 self.respond_read_index(&mut read);
-            } else if self.ready_to_handle_unsafe_replica_read(read.read_index.unwrap()) {
+            } else if self.ready_to_handle_unsafe_replica_read(read_index) {
                 self.respond_replica_read(&mut read);
+            } else if self.storage().apply_state().get_applied_index() + 100 <= read_index {
```

Comment:
Better to make the threshold 100 configurable.

Reply:
Also you can compare `self.storage().last_index()` to check if leader needs to send raft logs first.

Reply:
> 
It can't weok well if the apply work is slow. In the network test, the apply duration is slow after cancel the network partition error. So it will need to wait too long.
<img width="917" alt="image" src="https://github.com/tikv/tikv/assets/23159587/05029023-3d6f-4320-9fba-35fe3f0d014c">